pkgs/development/compilers/ghc/common-hadrian.nix

```diff
@@ -340,7 +340,14 @@ let
     ]
     ++ lib.optionals targetPlatform.useAndroidPrebuilt [
       "*.*.ghc.c.opts += -optc-std=gnu99"
-    ];
+    ]
+    # Inform GHC that we can't load dynamic libraries which forces iserv-proxy to load static libraries
+    ++
+      lib.optionals
+        (targetPlatform.isAndroid || (targetPlatform.isMusl && hostPlatform != targetPlatform))
+        [
+          "*.ghc.cabal.configure.opts += --flags=-dynamic-system-linker"
```

Comment:
The condition here is wrong, according to my experiments. Cross with musl works fine, but once we do `isStatic` it doesn't. Quite obviously.

I'm not sure whether we should just check `isStatic` - or whether we should actually check `enableShared`, which includes `isStatic` and more.

Given that the error message for android is:

```
qemu-aarch64: Could not open '/system/bin/linker64': No such file or directory
```

... I wonder whether we should actually add the `isAndroid` condition to `enableShared` to begin with?